### PR TITLE
Ported PartitionerTest to EE2

### PR DESCRIPTION
**Pull request by gcatron**
Summary: The PR ports the PartitionerTest to the new EE2

Documentation:

Progress on #3239 

Test Plan: verify everything builds and that PartitionerTest runs and passes.

### CODE REVIEW

**Comment by gcatron** [comment]
@beicy I rebased this PR and had to merge it with newer commits, can you verify I didn't accidentally remove your latest changes in the merge?

**Comment by nickgg** [comment]
Why are you doing every test on two EE's now?

**Comment by gcatron** [comment]
> Why are you doing every test on two EE's now?

Many of the tests involve some compile time manipulation of the graph. Since compile is now a module wide event this means we need two modules->two EEs. In this specific case, one EE to run the function without partitioning (EER). The second EE has it's function partitioned and then runs the partitioned subfunctions(EEP).

**Comment by nickgg** [comment]
Makes sense, can we add that in a comment?

File: tests/unittests/PartitionerTest.cpp [hunk 1]
```diff
@@ -410,10 +449,11 @@ TEST_F(PartitionerTest, Basic1Roofline) {
...
   };
-  for (auto const &p : nodeNamesMap) {
-    auto *N = p.first;
-    EXPECT_EQ(myPartitioner.getComputeTime(N), expectedComputeTime[p.second]);
-  }
+  // for (auto const &p : nodeNamesMap) {
+  //   auto *N = p.first;
+  //   EXPECT_EQ(myPartitioner.getComputeTime(N),
+  //   expectedComputeTime[p.second]);
+  // }
```
Comment (beicy) [inline tests/unittests/PartitionerTest.cpp]: I think this is part is testing the roofline. Why do you comment it out?

File: tests/unittests/PartitionerTest.cpp [hunk 2]
```diff
@@ -435,13 +475,13 @@ TEST_F(PartitionerTest, Basic1Roofline) {
...
       {"fc_add_bias4", 32},
   };
-  for (auto const &p : nodeNamesMap) {
-    auto *N = p.first;
-    EXPECT_EQ(myPartitioner.getMemUsage(N), expectedMemUsage[p.second]);
-  }
+  // for (auto const &p : nodeNamesMap) {
+  //   auto *N = p.first;
+  //   EXPECT_EQ(myPartitioner.getMemUsage(N), expectedMemUsage[p.second]);
+  // }
```
Comment (beicy) [inline tests/unittests/PartitionerTest.cpp]: Same. I think the checking should be here.

File: tests/unittests/PartitionerTest.cpp [hunk 3]
```diff
@@ -175,67 +184,77 @@ TEST_F(PartitionerTest, Basic1) {
...
-  Partitioner myPartitioner(&mod_, devices, /* saturateHost */ true);
+  Partitioner myPartitioner(&EEP.getModule(), devices, /* saturateHost */ true);
   CompilationContext cctx;
   auto err = myPartitioner.Partition(cctx);
   EXPECT_FALSE(errToBool(std::move(err)));
   DAGListTy dagList = std::move(myPartitioner.getPartitionResult());
-  EXPECT_EQ(mod_.getFunctions().size(), 2);
-  EXPECT_EQ(dagList.size(), 1);
-  ASSERT_TRUE(checkSaveNode(mod_));
+  ASSERT_EQ(EEP.getModule().getFunctions().size(), 2);
```
Comment (beicy) [inline tests/unittests/PartitionerTest.cpp]: I don't think this check is "fatal". Even it fails, we can still check the dagList.size() and then the saveNode. So I still prefer "EXPECT_EQ" here. I think the last check before running the DAG (i.e. checkSaveNode(EEP.getModule())) should be ASSERT_TRUE. But the checks before that (i.e. the ones checking the dag structure) could be EXPECT_TRUE.

File: tests/unittests/PartitionerTest.cpp (hunk 1 above)
Comment (gcatron) [reply tests/unittests/PartitionerTest.cpp]: fixed now

File: tests/unittests/PartitionerTest.cpp (hunk 2 above)
Comment (gcatron) [reply tests/unittests/PartitionerTest.cpp]: fixed too.

File: tests/unittests/PartitionerTest.cpp [hunk 4]
```diff
@@ -246,145 +266,174 @@ TEST_F(PartitionerTest, Basic2) {
...
+  bindings_.clear();
+  bindings_.allocate(EER.getModule().getPlaceholders());
+  updateInputPlaceholders2(bindings_, {bindings_.getPlaceholderByName("input")},
+                           {&in});
+  EER.run(bindings_);
+
+  // Since the partitioner will look at all nodesin the function post
+  // optimization and lowering, we need to do so here for the same list of
+  // nodes.
+  auto backend = createBackend(EEP.getBackendName());
```
Comment (beicy) [inline tests/unittests/PartitionerTest.cpp]: memory leak?

**Pull request closed without merging**